src/main/java/org/openstreetmap/atlas/utilities/caching/strategies/SystemTemporaryFileCachingStrategy.java

```diff
@@ -0,0 +1,123 @@
+            return Optional.empty();
+        }
+
+        final Path temporaryDirectory = Paths.get(TEMPORARY_DIRECTORY_STRING);
+        final String cachedFileName = this.getUUIDForResourceURI(resourceURI).toString();
+        final Path cachedFilePath = Paths.get(temporaryDirectory.toString(), cachedFileName);
+
+        final File cachedFile = new File(cachedFilePath.toString());
+        attemptToCacheFileLocally(cachedFile, defaultFetcher, resourceURI);
+
+        // cache hit!
+        if (cachedFile.exists())
+        {
+            logger.trace("Cache hit on resource {}, returning local copy", resourceURI);
+            return Optional.of(cachedFile);
+        }
+
+        logger.warn("Unexpected cache miss on resource {}", resourceURI);
+        return Optional.empty();
+    }
+
+    @Override
+    public String getName()
+    {
+        return "SystemTemporaryFileCachingStrategy";
+    }
+
+    @Override
+    public void invalidate()
+    {
+        throw new UnsupportedOperationException("Operation not supported at this time.");
+    }
+
+    private void attemptToCacheFileLocally(final File cachedFile,
+            final Function<URI, Resource> defaultFetcher, final URI resourceURI)
+    {
+        if (!cachedFile.exists())
+        {
+            logger.trace("Attempting to cache resource {} in temporary file {}", resourceURI,
+                    cachedFile.toString());
+
+            final Resource resourceFromDefaultFetcher = defaultFetcher.apply(resourceURI);
+            final File temporaryLocalFile = File.temporary();
+            try
+            {
+                resourceFromDefaultFetcher.copyTo(temporaryLocalFile);
+            }
+            catch (final Exception exception)
+            {
+                logger.error("Something went wrong copying {} to temporary local file {}",
+                        resourceFromDefaultFetcher.toString(), temporaryLocalFile.toString());
+                return;
+            }
+
+            // now that we have pulled down the file to a unique temporary location, attempt to
+            // atomically move it to the cache after re-checking for existence
+            if (!cachedFile.exists())
+            {
+                try
+                {
+                    final Path temporaryLocalFilePath = Paths.get(temporaryLocalFile.getPath());
+                    final Path cachedFilePath = Paths.get(cachedFile.getPath());
+                    Files.move(temporaryLocalFilePath, cachedFilePath,
+                            StandardCopyOption.ATOMIC_MOVE);
+                }
+                catch (final FileAlreadyExistsException exception)
+                {
+                    logger.trace("File {} is already cached", cachedFile.toString());
+                    return;
+                }
+                catch (final Exception exception)
+                {
+                    logger.error("Something went wrong moving {} to {}",
```

Comment:
Let's add the exception as the last parameter to the logger call, so we don't lose the stack trace if this ever happens.